COOKBOOK.md

```diff
@@ -0,0 +1,200 @@
+# Cookbook
+
+#### What is the Ion Object Mapper
+
+The Ion Object Mapper is a convenience library built to assist developers when dealing with data that's in the format of [Amazon Ion](https://amzn.github.io/ion-docs/). It is used to easily transform C# classes and objects into Ion format and vice-versa. There are numerous additional settings and features that can be configured for advanced users.
+
+#### Basic Serialization and Deserialization
+
+The mapper (IonSerializer) supports by default the types listed in the [spec](SPEC.md#primitive-type-conversion) under the "Primitive type conversion" section. There are ways to customize the conversion but the default use case adheres to the spec.
+
+```c#
+public class MyPOCO
+{
+    public string ValueOne { get; init; }
+    public int ValueTwo { get; init; }
+}        
+
+private static void SerializationExample()
+{
+    IonSerializer serializer = new IonSerializer();
+
+    MyPOCO poco = new MyPOCO
+    {
+        ValueOne = "one",
+        ValueTwo = 2,
+    };
+
+    Stream serialized = serializer.Serialize(poco);
+    // Most AWS libraries will take in a Stream type for using Ion with SDKs.
+    // 
+    // Alternatively, the Ion library can build an IIonReader from a Stream.
+    // An IIonReader can then be used to convert the Stream into IIonValue classes for use directly as a C# type.
+}
+
+private static void DeserializationExample(Stream serialized)
+{
+    // Assume serialized is the same stream from the SerializationExample.
+    IonSerializer serializer = new IonSerializer();
+
+    // The format of the serialized data must be known and a supported output type must be supplied to the Deserialize method.
+    MyPOCO poco = serializer.Deserialize<MyPOCO>(serialized);
+    // This poco will have the same values as the poco from the SerializationExample.
```

Comment:
Can you make SerializationExample return the stream, and have code in another method which calls both and passes the stream into DeserializationExample, as otherwise this comment is not quite accurate.